crates/muvm/src/guest/hidpipe.rs

```diff
@@ -119,7 +122,24 @@ fn init_uinput(sock: &mut UnixStream, user_id: u32) -> (u64, UInputHandle<File>)
         })
         .unwrap();
     uinput.dev_create().unwrap();
-    chown(uinput.evdev_path().unwrap(), Some(user_id), Some(0)).unwrap();
+    let ev_path = CString::new(
+        uinput
+            .evdev_path()
+            .unwrap()
+            .into_os_string()
+            .into_encoded_bytes(),
+    )
+    .unwrap();
+    match unsafe { fork().unwrap() } {
```

Comment:
Since this is the same pattern as for drop_caches, maybe extract this into a `run_as_root()` function that takes a closure?

Also `SAFETY` comments needed.